functions/src/list_index.ts

```diff
@@ -62,71 +62,81 @@ export const index_list = functions.runWith({timeoutSeconds:540}).https
 export const index_list2 = functions.runWith({timeoutSeconds:540}).https
 .onRequest(async (request, res) => {
   cors(request, res,async () => {
-
     const listId:string=request.query.list as string;
-    const fieldId:string=request.query.field as string;
-    const settings:any=request.body;
-
-    console.log(`indexing list ${listId} by ${fieldId} with settings ${JSON.stringify(settings)}`)
-    const items=
-    await db.collection("list").doc(listId).collection('item').get();
-
+    let indices = await db.collection('index').where('listId', '==', listId).get();
+    indices.forEach(index => {
```

Comment:
The problem with this code is that it is async and will start running in parallel while the rest of the code will be running at the same time. We have to first delete the docs and only then proceed to creating the new index.

Reply:
I have followed your sample code to use batch delete. Please review.